main.tf

```diff
@@ -8,11 +8,11 @@ module "ssh-key" {
 }
 
 resource "azurerm_kubernetes_cluster" "main" {
-  name                    = "${var.prefix}-aks"
+  name                    = var.cluster_name == "" ? "${var.prefix}-aks" : var.cluster_name
   kubernetes_version      = var.kubernetes_version
   location                = data.azurerm_resource_group.main.location
   resource_group_name     = data.azurerm_resource_group.main.name
-  dns_prefix              = var.prefix
+  dns_prefix              = var.prefix ? var.prefix : var.cluster_name
```

Comment:
`dns_prefix              = var.prefix` would be fine. It's weird to define two priority differently.